### PR TITLE
fix: update job name for manual cronjob creation

### LINKED ISSUE #18846: CronJob "Create Job" may render too long Job names
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When starting a Job from a CronJob in the ArgoCD UI, the Job is created by appending the date string `%Y%m%d%H%M` to the CronJob `.metadata.name`.

(See https://github.com/argoproj/argo-cd/blob/master/resource_customizations/batch/CronJob/actions/create-job/action.lua#L39)

This adds **13** characters to the `CronJob` name to create the `Job` object. However, according to the Kubernetes [CronJob spec](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/), when Kubernetes creates a `Job` from a `CronJob`, it will only add **11** characters, so the `CronJob` can be up until **52** characters long.

When creating a `Job` through ArgoCD UI "Create Job" option will produce the following error:

```
Unable to execute resource action: Job.batch "***-202406281143" is invalid: [metadata.labels: Invalid value: "***-202406281143": must be no more than 63 characters, spec.template.labels: Invalid value: "***-202406281143": must be no more than 63 characters]
```

Which makes sense, when the `CronJob` is 52 characters long, ArgoCD will create a `Job` with a name of 65 characters long, which will produce the above error.

**To Reproduce**

* Create an Application in ArgoCD that creates a `CronJob`, with a name of length 51-52
* Manually create a Job from that CronJob using the UI

image

This will show the above error.

**Expected behavior**

A Job will be started from the CronJob immediately with a correct name.

**Recommendations**

Replacing `%Y` (4-digit year) in the date format to `%y` (2-digit year), will create a Job with CronJob name +11 characters (which is conveniently _exactly_ within spec).

So, changing:

* `%Y%m%d%H%M` to
* `%y%m%d%H%M`

in the following file: https://github.com/argoproj/argo-cd/blob/master/resource_customizations/batch/CronJob/actions/create-job/action.lua#L39

**Version**

(server version only, commandline not available)

```json
{
    "Version": "v2.11.3+3f344d5",
    "BuildDate": "2024-06-06T08:42:00Z",
    "GitCommit": "3f344d54a4e0bbbb4313e1c19cfe1e544b162598",
    "GitTreeState": "clean",
    "GoVersion": "go1.21.9",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KustomizeVersion": "v5.2.1 2023-10-19T20:13:51Z",
    "HelmVersion": "v3.14.4+g81c902a",
    "KubectlVersion": "v0.26.11",
    "JsonnetVersion": "v0.20.0"
}
```

### PR DESCRIPTION
**Pull request by reiniertimmer**
Changes the name of Job resources when manually starting a CronJob from ArgoCD to be in line with the Kubernetes [CronJob spec](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/).

Fixes #18846

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by ishitasequeira — COMMENTED** [review]
Thanks @reiniertimmer !! Can we also add a brief section about this in https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/upgrading/2.12-2.13.md for users who are relying on the date in the name?

**Review by crenshaw-dev — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Good call re: release notes. Holding off merging. :-)

File: docs/operator-manual/upgrading/2.12-2.13.md [hunk 1]
```diff
@@ -45,3 +45,9 @@ Dex [v2.39.0](https://github.com/dexidp/dex/releases/tag/v2.39.0) included a bre
 > The validation of username and password in the LDAP connector is much more strict now.
 > As of today, Dex uses the EscapeFilter function to check for special characters in credentials and prevent injections by denying such requests.
 
+## Updated Job name for manually started CronJob jobs
+
+The naming of Jobs that are manually started from CronJobs (using ArgoCD) was changed. Instead of the previous postfix `-YYYYMMDDHHmm` (4-digit year), manually started Jobs now receive postfix `-YYMMDDHHmm` (2-digit year).
+
+The format of Jobs that are started from a CronJob on schedule (by Kubernetes) is not handled by ArgoCD and remains unchanged.
```
[the remark this answers is not in the document]
Comment (reiniertimmer) [reply docs/operator-manual/upgrading/2.12-2.13.md]: Thanks for the review @crenshaw-dev, committed your suggestions

**Review by crenshaw-dev — APPROVED** [review]

**Review by christianh814 — COMMENTED** [review]
Awesome!

**Pull request merged**